lux-lib/src/dependencies.rs

```diff
@@ -0,0 +1,72 @@
+use serde::{Deserialize, Serialize};
+use std::path::PathBuf;
+use which::which;
+
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct DependencyStatus {
+    pub name: String,
+    pub found: bool,
+    pub path: Option<PathBuf>,
+    pub version: Option<String>,
+}
+
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct DependencyReport {
+    pub c_compiler: DependencyStatus,
+    pub make: DependencyStatus,
+    pub cmake: DependencyStatus,
+    pub cargo: DependencyStatus,
+    pub pkg_config: DependencyStatus,
+}
```

Comment:
suggestion: 

1. Instead of a `found` flag, we can make invalid states irrepresentable.
2. Naming: 
  - `Tool` instead of `DependencyStatus`.
  - `ToolchainReport` instead of `DependencyReport`
3. As these types are part of the lux-lib API, we should avoid exposing the fields and instead expose getters.

```rust
pub struct Tool {
    name: String,
    version: Option<String>,
    path: PathBuf,
}

impl Tool {
    name(&self) -> &String {
        &self.name
    }
    // ...
}

pub struct ToolchainReport {
    c_compiler: Option<Tool>,
    make: Option<Tool>,
    cmake: Option<Tool>,
    cargo: Option<Tool>,
    pkg_config: Option<Tool>,
}

impl ToolchainReport {
    c_compiler(&self) -> Option<&Tool> {
        self.c_compiler.as_ref()
    }
    // ...
}
```